test/test_torch.py

```diff
@@ -2363,39 +2363,263 @@ def _test_rpow(self, cast):
     def test_rpow(self):
         self._test_rpow(self, lambda x: x)
 
+    @staticmethod
+    def _check_against_np(self, tensor, exp):
+        tensor_np = tensor.cpu().numpy()
+        exp_np = exp if isinstance(exp, int) else exp.cpu().numpy()
+        expected = torch.LongTensor(tensor_np ** exp_np).type_as(tensor)
+        self.assertEqual(torch.pow(tensor, exp), expected)
+        self.assertEqual(tensor.pow(exp), torch.pow(tensor, exp))
+
     @staticmethod
     def _test_int_pow(self, cast):
         if not TEST_NUMPY:
             return
 
-        def check_against_np(tensor, exp):
-            tensor_np = tensor.cpu().numpy()
-            exp_np = exp if isinstance(exp, int) else exp.cpu().numpy()
-            expected = torch.LongTensor(tensor_np ** exp_np).type_as(tensor)
-            self.assertEqual(torch.pow(tensor, exp), expected)
-            self.assertEqual(tensor.pow(exp), torch.pow(tensor, exp))
-
         typecasts = [
             lambda x: x.long(),
-            lambda x: x.short(),
-            lambda x: x.byte(),
         ]
 
         if not IS_WINDOWS:
             typecasts.append(lambda x: x.int())
 
-        shape = (11, 5)
-        tensor = cast(torch.LongTensor(shape).random_(-10, 10))
-        exps = [0, 1, 2, 5, cast(torch.LongTensor(shape).random_(0, 20))]
+        tensor = cast(torch.LongTensor(3, 3).random_(-10, 10))
```

Comment:
Ditto, it would be nice to use the `torch.tensor` ctor

Reply:
addressed